kimchi/src/circuits/polynomials/foreign_field_add/circuitgates.rs

```diff
@@ -138,7 +137,13 @@ where
     const CONSTRAINTS: u32 = 5;
 
     fn constraint_checks<T: ExprOps<F>>(env: &ArgumentEnv<F, T>) -> Vec<T> {
-        let foreign_modulus: [T; LIMB_COUNT] = array::from_fn(|i| env.coeff(i));
+        let foreign_modulus: [T; LIMB_COUNT] = [env.coeff(0), env.coeff(1), env.coeff(2)];
```

Comment:
why switch from `array::from_fn`?

Reply:
Oh because I was not sure it would be correctly parsed when the coefficients are not just 3 but 4. Meaning, I was not sure which 3 it would get to fit the LIMB_COUNT length. But I can revert that if it works fine.